linker/Linker.Steps/MarkStep.cs

```diff
@@ -1587,5 +1597,143 @@ protected virtual void MarkInterfaceImplementation (TypeDefinition type, Interfa
 			MarkCustomAttributes (iface);
 			MarkType (iface.InterfaceType);
 		}
+
+
+		void MarkSomethingUsedViaReflection (string reflectionMethod, Action<Collection<Instruction>, string, TypeDefinition, BindingFlags> markMethod, Collection<Instruction> instructions)
+		{
+			for (var i = 0; i < instructions.Count; i++) {
+				var instruction = instructions [i];
+				if (instruction.OpCode != OpCodes.Call && instruction.OpCode != OpCodes.Callvirt)
+					continue;
+
+				var methodBeingCalled = instruction.Operand as MethodReference;
+				if (methodBeingCalled == null || methodBeingCalled.DeclaringType.Name != "Type" || methodBeingCalled.DeclaringType.Namespace != "System")
+					continue;
+
+				if (methodBeingCalled.Name != reflectionMethod)
+					continue;
+
+				_context.Tracer.Push ($"Reflection-{methodBeingCalled}");
+				var nameOfThingUsedViaReflection = OperandOfNearestInstructionBefore<string> (i, OpCodes.Ldstr, instructions);
+				var bindingFlags = (BindingFlags) OperandOfNearestInstructionBefore<sbyte> (i, OpCodes.Ldc_I4_S, instructions);
```

Comment:
You could use `BindingFlags` instead of `<sbyte>`

Reply:
This is not possible, since the operand for the `Ldc_I4_S` opcode is an `sbyte`. We need to find the `sbyte`, then cast the result to `BindingFlags`.

Reply:
It does only casts of the result so `OperandOfNearestInstructionBefore<BindingFlags> (i, OpCodes.Ldc_I4_S, instructions);` should do exactly same

Reply:
Now looking at this again i4_s will be used only for low values, if you write a sample with `ExactBinding` you'd need to tweak this code

Reply:
I don't think we need to worry about `ExactBindings`. Maybe that is something we could add in the future, but I'd only like to support limited bindings flags now.